x-pack/plugins/ml/public/application/explorer/hooks/use_selected_cells.ts

```diff
@@ -5,21 +5,21 @@
  */
 
 import { useCallback, useMemo } from 'react';
-import { useUrlState } from '../../util/url_state';
 import { SWIMLANE_TYPE } from '../explorer_constants';
 import { AppStateSelectedCells } from '../explorer_utils';
+import { ExplorerAppState } from '../../../../common/types/ml_url_generator';
 
 export const useSelectedCells = (
-  appState: any,
-  setAppState: ReturnType<typeof useUrlState>[1]
+  appState: ExplorerAppState,
+  setAppState: (update: Partial<ExplorerAppState>) => void
 ): [AppStateSelectedCells | undefined, (swimlaneSelectedCells: AppStateSelectedCells) => void] => {
   // keep swimlane selection, restore selectedCells from AppState
   const selectedCells = useMemo(() => {
     return appState?.mlExplorerSwimlane?.selectedType !== undefined
       ? {
           type: appState.mlExplorerSwimlane.selectedType,
-          lanes: appState.mlExplorerSwimlane.selectedLanes,
-          times: appState.mlExplorerSwimlane.selectedTimes,
+          lanes: appState.mlExplorerSwimlane.selectedLanes!,
+          times: appState.mlExplorerSwimlane.selectedTimes!,
```

Comment:
Maybe it's worth doing a type guard to get rid of the `!`s? The code should work fine as is, but it's a bit intransparent because you'd need to know the interface to understand why the `!`s are ok.

Reply:
@walterra I actually wanted to change the interface itself, but it's going to be too many changes for this PR.
Correct me if I'm wrong, but it seems like `mlExplorerSwimlane` should be optional, but when it's defined, the nested props should be required.

Reply:
Sounds good to improve in a follow up! You're right about the structure!